beacon-chain/blockchain/chain_info.go

```diff
@@ -163,7 +163,7 @@ func (s *Service) HeadState(ctx context.Context) (*state.BeaconState, error) {
 		return s.headState(ctx), nil
 	}
 
-	return s.beaconDB.HeadState(ctx)
+	return s.stateGen.StateByRoot(ctx, s.headRoot())
```

Comment:
We need to be careful here, given that this is the main method that all services access our head state. If we ever come to this point, shouldn't we recache the state ? So that we dont miss the next time we call for our head state.